trieste/acquisition/optimizer.py

```diff
@@ -168,11 +169,63 @@ def optimize_discrete(
     return _get_max_discrete_points(points, target_func)
 
 
+InitialPointSampler = Callable[[SearchSpace, int, int], TensorType]
+"""
+Type alias for a function that generates initial samples for an optimizer.
+Takes a search space, the number of samples to generate, and an offset, and returns
+that many samples (at the offset).
+"""
+
+
+def sample_from_space(space: SearchSpace, num_samples: int, offset: int) -> TensorType:
+    """Default initial point sampler that samples randomly from the space."""
+    return space.sample(num_samples)
+
+
+def sample_from_sequence(sequence: Sequence[TensorType]) -> InitialPointSampler:
+    """
+    Initial point sampler that returns points from a prebuilt sequence (e.g. a list or Tensor).
+    Raises a :exc:`ValueError` if there are insufficiently many points to return.
+    """
+
+    def _sampler(space: SearchSpace, num_samples: int, offset: int) -> TensorType:
+        if len(sequence) < offset + num_samples:
+            raise ValueError(
+                f"Insufficient samples ({offset+num_samples} required, {len(sequence)} available"
+            )
+        samples = sequence[offset : offset + num_samples]
+        if not isinstance(slice, tf.Tensor):
+            samples = tf.concat(samples, axis=0)
+        return samples
+
+    return _sampler
+
+
+def sample_from_iterator(iterator: Iterator[TensorType]) -> InitialPointSampler:
+    """
+    Initial point sampler that returns points from a prebuilt iterator (e.g. a generator).
+    Raises a :exc:`ValueError` if the iterator is exhausted with insufficiently many points.
+    """
+
+    def _sampler(space: SearchSpace, num_samples: int, offset: int) -> TensorType:
+        slice = itertools.islice(iterator, num_samples)
+        samples = tf.concat(list(slice), axis=0)
+        if len(samples) < num_samples:
+            raise RuntimeError(
+                f"Insufficient samples ({num_samples} requested, {len(samples)} available"
+            )
+        return samples
+
+    return _sampler
+
+
 def generate_continuous_optimizer(
     num_initial_samples: int = NUM_SAMPLES_MIN,
     num_optimization_runs: int = 10,
     num_recovery_runs: int = 10,
     optimizer_args: Optional[dict[str, Any]] = None,
+    split_initial_samples: Optional[int] = 100_000,
```

Comment:
That is a much better idea! I've pushed a first attempt at this but will see what breaks and write some tests before asking for a review.

As an aside, it's actually possible to do this without breaking the interface by still letting people specify an int for `num_initial_samples` if they want to. There's an argument that we should rename the argument from `num_initial_samples` to `initial_samples`, but I'm worried that this would break a _lot_ of code (and not just ours). Another option is to add a separate `initial_samples` argument and make people specify just one of this and `num_initial_samples`.